### PR TITLE
add String Type to drop/takeWhile API

### PR DESCRIPTION
**Pull request by adispring**

### CODE REVIEW

File: src/dropWhile.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 var _curry2 = require('./internal/_curry2');
 var _dispatchable = require('./internal/_dispatchable');
 var _xdropWhile = require('./internal/_xdropWhile');
+var slice = require('./slice');
```
Comment (buzzdecafe) [inline src/dropWhile.js]: i believe the internal references were deliberately removed to simplify the dependency tree?

Comment (adispring) [inline src/dropWhile.js]: But if using Array.prototype.slice, the returned value will always be an array. Then how to let dropWhile compatible with String?

Comment (buzzdecafe) [inline src/dropWhile.js]: Yes you are correct of course, my apologies for being dumb. I blame the eclipse

Comment (CrossEye) [inline src/dropWhile.js]: In fact, it's the year of the eclipse.  We can use that excuse until January, I believe. :smile:

**Comment by CrossEye** [comment]
:+1:

**Comment by buzzdecafe** [comment]
:cow2:

File: src/dropWhile.js (hunk 1 above)
Comment (adispring) [reply src/dropWhile.js]: A big and long eclipse that hide something 😁, thanks for merging.

**Pull request merged**